src/js/ripple/orderbook.js

```diff
@@ -176,80 +212,200 @@ OrderBook.prototype.unsubscribe = function() {
+    }
+
+    self._subscribed = true;
+
+    callback(null, res);
+  };
+
+  function handleError(err) {
+    if (self._remote.trace) {
+      log.info('failed to subscribe to transactions', self._key, err);
+    }
+
+    callback(err);
+  };
+
+  var request = this._remote.requestSubscribe();
+  request.addStream('transactions');
+  request.once('success', handleSubscribed);
+  request.once('error', handleError);
+  request.request();
+
+  return request;
+};
+
+/**
+ * Reset cached owner's funds, offer counts, and offer sums
  */
 
 OrderBook.prototype.resetCache = function() {
-  this._ownerFunds = { };
-  this._ownerOffers = { };
-  this._offerCounts = { };
+  this._ownerFunds = {};
+  this._ownerOffersTotal = {};
+  this._offerCounts = {};
   this._synchronized = false;
 };
 
 /**
- * Check that the funds for offer owner have been cached
+ * Check whether owner's funds have been cached
  *
- * @param {String} account address
+ * @param {String} account - owner's account address
  */
 
-OrderBook.prototype.hasCachedFunds = function(account) {
+OrderBook.prototype.hasOwnerFunds = function(account) {
   assert(UInt160.is_valid(account), 'Account is invalid');
   return this._ownerFunds[account] !== void(0);
 };
 
 /**
- * Add cached offer owner funds
+ * Set owner's, transfer rate adjusted, funds in cache
  *
- * @param {String} account address
- * @param {String} funded amount
+ * @param {String} account - owner's account address
+ * @param {String} fundedAmount
  */
 
-OrderBook.prototype.addCachedFunds = function(account, fundedAmount) {
+OrderBook.prototype.setOwnerFunds = function(account, fundedAmount) {
   assert(UInt160.is_valid(account), 'Account is invalid');
   assert(!isNaN(fundedAmount), 'Funded amount is invalid');
-  this._ownerFunds[account] = fundedAmount;
+
+  this._ownerFundsUnadjusted[account] = fundedAmount;
+  this._ownerFunds[account] = this.applyTransferRate(fundedAmount);
 };
 
 /**
- * Get cached offer owner funds
+ * Get owner's cached, transfer rate adjusted, funds
  *
- * @param {String} account address
+ * @param {String} account - owner's account address
+ * @return {String}
  */
 
-OrderBook.prototype.getCachedFunds = function(account) {
+OrderBook.prototype.getOwnerFunds = function(account) {
   assert(UInt160.is_valid(account), 'Account is invalid');
   return this._ownerFunds[account];
 };
 
 /**
- * Remove cached offer owner funds
+ * Get owner's cached unadjusted funds
+ *
+ * @param {String} account - owner's account address
+ * @return {String}
+ */
+
+OrderBook.prototype.getUnadjustedOwnerFunds = function(account) {
+  assert(UInt160.is_valid(account), 'Account is invalid');
+  return this._ownerFundsUnadjusted[account];
+};
+
+/**
+ * Remove cached owner's funds
  *
- * @param {String} account address
+ * @param {String} account - owner's account address
  */
 
-OrderBook.prototype.removeCachedFunds = function(account) {
+OrderBook.prototype.deleteOwnerFunds = function(account) {
```

Comment:
"delete" is more technically loaded; there is a `delete` keyword in JS (which isn't used here), etc.